Source/WebCore/html/HTMLDetailsElement.cpp

```diff
@@ -155,15 +160,27 @@ void HTMLDetailsElement::attributeChanged(const QualifiedName& name, const AtomS
             RefPtr root = shadowRoot();
             ASSERT(root);
             if (!newValue.isNull()) {
-                root->appendChild(*m_defaultSlot);
+                if (document().settings().detailsAutoExpandEnabled()) {
+                    m_defaultSlot->removeInlineStyleProperty(CSSPropertyContentVisibility);
+                    m_defaultSlot->removeInlineStyleProperty(CSSPropertyDisplay);
+                    if (auto* renderer = this->renderer())
+                        renderer->repaint(); // Repaints disclosure triangle
+                } else
+                    root->appendChild(*m_defaultSlot);
                 queueDetailsToggleEventTask(DetailsState::Closed, DetailsState::Open);
                 if (document().settings().detailsNameAttributeEnabled() && !attributeWithoutSynchronization(nameAttr).isEmpty()) {
                     ShouldNotFireMutationEventsScope scope(document());
                     for (auto& otherDetailsElement : otherElementsInNameGroup())
                         otherDetailsElement->removeAttribute(openAttr);
                 }
             } else {
-                root->removeChild(*m_defaultSlot);
+                if (document().settings().detailsAutoExpandEnabled()) {
+                    m_defaultSlot->setInlineStyleProperty(CSSPropertyContentVisibility, CSSValueHidden);
+                    m_defaultSlot->setInlineStyleProperty(CSSPropertyDisplay, CSSValueBlock);
+                    if (auto* renderer = this->renderer())
```

Comment:
```suggestion
                    if (RefPtr renderer = this->renderer())
```

Reply:
renderer is not ref-counted. So what you actually want is:
```
if (CheckedPtr renderer = this->renderer())
```

Reply:
Thanks both — changed to using `CheckedPtr`